### ISSUE TITLE
Installable wev app / iPhone app / Android app interest tracking. 

### ISSUE THREAD

**Issue by jamesread** [OliveTin/OliveTin#31]
OliveTin's protocol makes an app fairly straightforward, but I'm not personally a big fan of apps. 

However, if you're interested, register that interest here. It would have to get quite a bit of demand / good use cases though before I personally consider it. 

1. What version would you need? Installable web app, iPhone app, or Android app?
2. Would just actions on the app be enough? 
3. Is an embedded browser ok, or does it have to be native buttons?
4. Why does your use case need an app instead of the webui?

**Comment by ashleycawley** [comment]
OliveTin offers a simple web based UI with large buttons, whilst I personally haven't yet tried accessing it on mobile yet I'm guessing it displays quite nicely on there. Platforms like iOS and Android offer ways to "Bookmark" a favourite page to your Home Screen, I imagine this would be a good quick solution for those looking to use it on mobile? Take a peak at  [this page](https://www.macrumors.com/how-to/add-a-web-link-to-home-screen-iphone-ipad/) as an example of what I'm talking about. 

Using that method would completely mitigate the need for native Apps to be maintained on various different platforms (iOS/Android) and it would also ensure that the visual style and experience would remain the same on both platforms.

I wonder if there could be some intelligent user-agent checking done by OliveTin when the page loads, so it knew if the user was visiting from a mobile device, then at the bottom of the page it could perhaps display a discrete tool-tip that read something like:

**_"Visiting from a mobile device? Click here to learn how to bookmark OliveTin to your home screen."_**

As I appreciate that not everyone will know that you can bookmark a URL/page to your home-screen or how to do it. The "click here" bit could be a link to a OliveTin doc which would describe how to bookmark OliveTin to the Home screen on common mobile platforms.

Just a thought.

**Comment by mill1000** [comment]
I agree that a [PWA](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps) would be sufficient. Even better if individual shortcuts could be installed to the home screen.

**Comment by kjboa** [comment]
I currently use OliveTin with a Shortcut on my phone, and I must say I would rather have a fully native solution. Even though it definitely is usable in it's current state and the buttons scale nicely the logs section doesn't: if you have a huge log output scrolling all the way to the side or the bottom of the page is not really a comfortable solution.

**Comment by ashleycawley** [comment]
> the logs section doesn't: if you have a huge log output scrolling all the way to the side or the bottom of the page is not really a comfortable solution

Unfortunately I think that's always going to be a problem on a mobile device, whether in a native App or not, if you're dealing with huge log outputs then its going to be hard to display it on a small mobile screen elegantly without you having to scroll. I mean it could word-wrap the output, perhaps have a optional tick-box for that, but you're still going to have to scroll whether its a native app or not, so the question is is it really worth someones time/effort to build and maintain native mobile Apps just to get around that?

**Comment by matthewstrasiotto** [comment]
Native apps are usually overkill, and the olivetin web-app is responsive & touch friendly.

A PWA **might** be nice, a native app that implements a web-wrapper / webview is probably overkill, a native app that actually re-implements the UI is **definitely** overkill.

@kjboa 

> the logs section doesn't [scale nicely]: if you have a huge log output scrolling all the way to the side or the bottom of the page is not really a comfortable solution.

Though this is a valid complaint, a native app isn't the only solution and is probably too much - This would be fairly easy to address with some stylesheet &/or markup modifications,

**Comment by EzequielBruni** [comment]
I certainly wouldn't mind having an app that's just a glorified browser-wrapper thing.

**Comment by edersong** [comment]
Follow an excellent guide to configure PWA (Progressive web app)
https://blog.heroku.com/how-to-make-progressive-web-app
